pkg/datapath/loader/base.go

```diff
@@ -444,13 +449,13 @@ func (l *loader) Reinitialize(ctx context.Context, tunnelConfig tunnel.Config, d
 			log.WithError(err).Fatal("failed to compile encryption programs")
 		}
 
-		if err := l.reinitializeIPSec(ctx); err != nil {
+		if err := l.reinitializeIPSec(); err != nil {
 			return err
 		}
 	}
 
 	if err := l.reinitializeOverlay(ctx, tunnelConfig); err != nil {
-		return err
+		log.WithError(err).Fatal("Failed to initialize overlay device")
```

Comment:
So here I wonder, why is it okay to go from returning the error to just logging it? Maybe you can break out this change (plus the errors.Join stuff in reinitializeIPSec above) into a separate commit with an explanation?

Reply:
Notice the use of `.Fatal()` here. I updated this to be in line with preceding statements, but I'll drop the change to leave it consistently inconsistent. No strong need to change this.

Reply:
If anything, we should probably remove most Fatal()s to allow the caller to recover, not add more of them.